### PR TITLE
[SPARK] Test UPDATE/MERGE/DELETE Commands

### PR DESCRIPTION
**Pull request by amaliujia**
**Description of changes**

Test UPDATE/MERGE/DELETE Commands

### CODE REVIEW

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 1]
```diff
@@ -152,6 +152,66 @@ public void testCredentialDelta() throws ApiException, IOException {
     session.stop();
   }
 
+  @Test
+  public void testDeleteDeltaTable() throws ApiException, IOException {
+    createCommonResources();
+    SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG, CATALOG_NAME);
```
Comment (cloud-fan) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:158]:
```suggestion
    SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG);
```

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 2]
```diff
@@ -152,6 +152,66 @@ public void testCredentialDelta() throws ApiException, IOException {
...
+
+    session.stop();
+  }
+
+  @Test
+  public void testMergeDeltaTable() throws ApiException, IOException {
+    createCommonResources();
+    SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG, CATALOG_NAME);
+
+    String loc0 = "s3://test-bucket0" + generateTableLocation(SPARK_CATALOG, DELTA_TABLE);
```
Comment (cloud-fan) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:177]: nit: let's use `loc1` to match `t1`

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 3]
```diff
@@ -152,6 +152,66 @@ public void testCredentialDelta() throws ApiException, IOException {
...
+    setupExternalDeltaTable(SPARK_CATALOG, DELTA_TABLE, loc0, new ArrayList<>(0), session);
+    String t1 = SPARK_CATALOG + "." + SCHEMA_NAME + "." + DELTA_TABLE;
+    session.sql("INSERT INTO " + t1 + " SELECT 1, 'a'");
+
+    String table2 = DELTA_TABLE + "2";
```
Comment (cloud-fan) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:182]: We already have `ANOTHER_PARQUET_TABLE`, can we add `ANOTHER_DELTA_TABLE`?

**Review by cloud-fan — APPROVED** [review]

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java (hunk 2 above)
Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:177]: This is not addressed

Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:178]: still not addressed...

Comment (amaliujia) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java]: I messed up some local commits and a commit is not pushed....

Now it should be good.

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 4]
```diff
@@ -152,6 +153,65 @@ public void testCredentialDelta() throws ApiException, IOException {
...
+
+    session.stop();
+  }
+
+  @Test
+  public void testUpdateDeltaTable() throws ApiException, IOException {
+    createCommonResources();
+    SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG);
+
+    String loc0 = "s3://test-bucket0" + generateTableLocation(SPARK_CATALOG, DELTA_TABLE);
```
Comment (cloud-fan) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:203]:
```suggestion
    String loc1 = "s3://test-bucket0" + generateTableLocation(SPARK_CATALOG, DELTA_TABLE);
```

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 5]
```diff
@@ -152,6 +153,65 @@ public void testCredentialDelta() throws ApiException, IOException {
...
+    setupExternalDeltaTable(SPARK_CATALOG, DELTA_TABLE, loc0, new ArrayList<>(0), session);
```
Comment (cloud-fan) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:204]:
```suggestion
    setupExternalDeltaTable(SPARK_CATALOG, DELTA_TABLE, loc1, new ArrayList<>(0), session);
```

**Review by cloud-fan — APPROVED** [review]

**Pull request merged**